### ISSUE TITLE
The .msi generated from the Install Toolkit can prevent itself from working

### ISSUE THREAD

**Issue by anthonypants** [OfficeDev/Office-IT-Pro-Deployment-Scripts#139]
The .msi installs the Microsoft Office 365 ProPlus Installer, which lives in "%ProgramFiles(x86)%\Microsoft Office 365 ProPlus Installer\". The files in this folder are an executable and an xml file.
One of the very first tasks generated by this executable is to stage some files under "%windir%\Temp\", after which it begins the installation process.
If this installation is run more than once, the installer will fail. Personally, I encountered this issue when I tried to install an x86 version, but the Office 365 installer complained about an x64 plugin that was already installed. So I moved an x64 installer over to this machine, and the ProPlus Installer would no longer work with either version. I've also tested using this installer on a fresh Windows 7 VM, installing Office 365 ProPlus successfully, uninstalling the product, and then attempting to use the ProPlus Installer a second time. The same issue was encountered.

The error that is generated by the "Office 365 ProPlus Installer", which does not appear to be logged anywhere, is:
`Extracting Install Files...ERROR: System.IO.IOException: Cannot create a file when that file already exists.

   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.File.InternalMove(String sourceFileName, String destFileName, Boolean checkHost)
   at InstallOffice.MoveFile(String rootDirectory, String md5Hash, String fileName)
   at InstallOffice.GetEmbeddedItems(String targetDirectory)
   at InstallOffice.Initialize()
   at InstallOffice.RunProgram()
   at InstallOffice.Main(String[] args)`
This error message shows up in a command prompt briefly, but can be seen if you open a command prompt first, and run the executable at "%ProgramFiles(x86)%\Microsoft Office 365 ProPlus Installer\InstallOfficeProPlus.exe".

Uninstalling the Microsoft Office 365 ProPlus Installer and reinstalling it does not clear out the files in the "%windir%\Temp\" directory. Running the "InstallOfficeProPlus.exe" executable with the "/uninstall" parameter also generates the above error, and also does not remove any files in the "%windir%\Temp\" directory.

After investigation with ProcMon, it turns out the files it's trying to overwrite are the staging files in the "%windir%\Temp\" directory.

The Microsoft Office 365 ProPlus installer should be able to either use these files, or it should be able to overwrite them instead of only checking if the target exists, and then silently failing.

The Microsoft Office 365 ProPlus Installer should also generate a log for errors such as these, as opening the .msi will very briefly display this error inside of a command prompt, and then the command prompt will close upon process termination due to the error. A "/log=[PATH]" parameter would also help. The only logs produced from this process are the logs generated by the actual Office 365 ProPlus installation.

**Comment by johnskydiver** [comment]
Thank you for your feedback.  The MSI tends to fail if you see more than one instance of "Microsoft Office 365 ProPlus Installer" in your programs and features.  It's best if you uninstall as many instances of that program as you see.  As far as running the executable from the path "%ProgramFiles(x86)%\Microsoft Office 365 ProPlus Installer\", yes that will work, but if you're having issues with it throwing errors from trying to copy over files in the windir temp directory, chances are that you are not running it with elevated privileges ie try running it as admin, either the exe itself, or run the command window as admin, and run the executable from there.  The program will overwrite the files in the windir temp folder, I've tested it.  Please try my suggestions here, and post whether or not they've worked.

Thanks.

John J.

**Comment by anthonypants** [comment]
You can't shift-click an .msi and run as admin.
Running msiexec on the file from an Administrator cmd window produces the same error, and does not overwrite files.

**Comment by johnskydiver** [comment]
I was not referring to running the MSI as admin, I was referring running the exe in the "%ProgramFiles(x86)%\Microsoft Office 365 ProPlus Installer\" path as admin.  I guess I didn't make that clear enough.  Did you go into add/remove programs and make sure all instances of "Microsoft Office 365 ProPlus Installer" were removed?  That would prevent a second launch of the MSI from working correctly.  The files in temp directory will be overwritten unless you're running it without admin privileges.  Sorry for the confusion.

**Comment by anthonypants** [comment]
If you uninstall the Microsoft Office 365 ProPlus Installer, all files in the directory at "%ProgramFiles(x86)%\Microsoft Office 365 ProPlus Installer" are removed. This makes it impossible to perform the sequences of events you're suggesting. Additionally, as reported earlier, if the Microsoft Office 365 ProPlus Installer is uninstalled, files in the directory at "%windir%\Temp" are not removed. So, uninstalling this product makes it impossible to reinstall without first clearing out the "%windir%\Temp" directory. This is the issue I have reported.

If the "%windir%\Temp" directory has the Office 365 ProPlus Installer files in it, opening the executable at "%ProgramFiles(x86)%\Microsoft Office 365 ProPlus Installer\InstallOfficeProPlus.exe" with administrator privileges generates the error as reported above. You can attempt to right-click the executable and select Run as Administrator or you can open a cmd prompt as administrator and run the executable, but each method generates this error.

**Comment by shaunep** [comment]
I am experiencing similar issues as to what @anthonypants is reporting.

Extracting Install Files...ERROR: System.IO.IOException: Cannot create a file when that file already exists.

   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.File.InternalMove(String sourceFileName, String destFileName, Boolean checkHost)
   at InstallOffice.MoveFile(String rootDirectory, String md5Hash, String fileName)
   at InstallOffice.GetEmbeddedItems(String targetDirectory)
   at InstallOffice.Initialize()
   at InstallOffice.RunProgram()
   at InstallOffice.Main(String[] args)

Here is what my configuration.xml file looks like:
```
<Configuration>
  <Add OfficeClientEdition="32" Channel="Deferred">
    <Product ID="O365ProPlusRetail">
      <Language ID="en-us" />
    </Product>
  </Add>
  <Updates Enabled="TRUE" Channel="Deferred" />
  <Display Level="Full" AcceptEULA="True" />
  <Property Name="PinIconsToTaskbar" Value="TRUE" />
  <Property Name="AUTOACTIVATE" Value="0" />
  <Property Name="FORCEAPPSHUTDOWN" Value="FALSE" />
  <Property Name="SharedComputerLicensing" Value="0" />
  <Logging Level="Off" />
</Configuration>
```

Also, on the generate section of the toolkit, I selected "Embed Office installation files".

I have tried both MSI and EXE as the install type and receive the same error in both cases. Of course for the MSI, it extracts the EXE and then throws the error. I am running the installers as administrator from the command prompt.
